packages/spanner/system-test/spanner.js

```diff
@@ -1034,9 +1035,56 @@ var spanner = new Spanner(env);
 
             transaction.commit(function(err) {
               assert.ifError(err);
+              firstTransaction.commit(assert.ifError);
```

Comment:
I dont understand why do you need to call commit here? On line 1013, instead of returning, you should go ahead with the commit and make sure that commit fails.
Also in the second attempt of first transaction, we should assert that we read the values that were inserted by second transaction.